### PR TITLE
Update what-is-fsharp.md

### PR DESCRIPTION
**Pull request by dsyme**
We have updated the messaging around F# per [the landing page here](https://dotnet.microsoft.com/languages/fsharp) and also on https://fsharp.org

This makes some changes on the "What is F#" page to help correspond to this.

This page is still not ideal but this is a useful start I think.  

cc @cartermp @BethMassi

### CODE REVIEW

File: docs/fsharp/what-is-fsharp.md [hunk 1]
```diff
@@ -1,31 +1,29 @@
 ---
 title: What is F#
-description: Learn about what the F# programming language is and what F# programming is like. Learn about rich data types, functions, and how they fit together.
+description: Learn about what the F# programming language is and what F# programming is like. Learn about rich types, functions, and how they fit together.
 ms.date: 08/03/2018
```
Comment (cartermp) [inline docs/fsharp/what-is-fsharp.md:4]: Please update the date

File: docs/fsharp/what-is-fsharp.md [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
 ---
 # What is F\#
 
-F# is a functional programming language that makes it easy to write correct and maintainable code.
+F# is an open-source, cross-platform, interoperable programming language for writing succinct, robust and performant code.
```
Comment (cartermp) [inline docs/fsharp/what-is-fsharp.md:8]: I think that, given this is docs and not a marketing site, mentioning functional here actually does make sense. The rest of the sentence does feel like the right framing though.

File: docs/fsharp/what-is-fsharp.md [hunk 3]
```diff
@@ -1,31 +1,29 @@
...
 
-F# programming primarily involves defining types and functions that are type-inferred and generalized automatically. This allows your focus to remain on the problem domain and manipulating its data, rather than the details of programming.
+F# programming is data-oriented and code involves transforming data with functions. An F# program is often
```
Comment (cartermp) [inline docs/fsharp/what-is-fsharp.md:10]:
```suggestion
F# programming is data-oriented, where code involves transforming data with functions. An F# program is often
```

File: docs/fsharp/what-is-fsharp.md [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
+a succinct definition of types that model a domain and functions that implement your chosen transformations.
+Your focus remains on your problem domain, rather than the details of programming.
```
Comment (cartermp) [inline docs/fsharp/what-is-fsharp.md:12]: Could this get broken up into some bullet points? I feel like that would be clearer than the paragraph

File: docs/fsharp/what-is-fsharp.md [hunk 5]
```diff
@@ -1,31 +1,29 @@
...
 // Defines a function that takes a name and produces a greeting.
-let getGreeting name = $"Hello, {name}! Isn't F# great?"
+let getGreeting name = $"Hello, {name}"
 
-[<EntryPoint>]
-let main args =
-    // Defines a list of names
-    let names = [ "Don"; "Julia"; "Xi" ]
+// Defines a list of names
+let names = [ "Peter"; "Julia"; "Xi" ]
```
Comment (cartermp) [inline docs/fsharp/what-is-fsharp.md:21]: Glad you left in one of my little easter eggs still 😉

File: docs/fsharp/what-is-fsharp.md (hunk 2 above)
Comment (dsyme) [reply docs/fsharp/what-is-fsharp.md:8]: It's the landing page for a lot of "What is F#" links so we would like this core description to be the start. We can then explain the technical characteristics.  

"Functional" will in general be de-emphasised across the docs slowly.  Data+Functions will be emphasised in replacement, per https://twitter.com/isaac_abraham/status/1417122528607227906/photo/1 :)

File: docs/fsharp/what-is-fsharp.md (hunk 5 above)
Comment (dsyme) [reply docs/fsharp/what-is-fsharp.md:21]: ha yes

File: docs/fsharp/what-is-fsharp.md (hunk 4 above)
Comment (dsyme) [reply docs/fsharp/what-is-fsharp.md]: I shortened this

**Review by cartermp — APPROVED** [review]

**Pull request merged**